File: quicksort.java
package com.test;

public class QuickSort {
	public static void main(String[] args) {
		QuickSort qs = new QuickSort();
		int[] a = new int[] { 6, 10, 13, 5, 8, 3, 2, 11 };
		qs.quickSort(a, 0, a.length - 1);
	}

	public int partition(int[] a, int p, int q) {
		int x = a[p];
		int i = p;
		for (int j = i + 1; j &lt;= q; j++) {
			if (a[j] &lt;= x) {
				i += 1;
				swap(a, i, j);
			}
		}
		swap(a, p, i);
		System.out.println("Array after iteration: ");
		printArray(a);
		return i;
	}

	public void quickSort(int[] a, int p, int q) {
		if (p &lt; q) {
			int r = partition(a, p, q);
			quickSort(a, p, r - 1);
			quickSort(a, r + 1, q);
		}

	}

	public void swap(int[] a, int i, int j) {
		int temp = a[i];
		a[i] = a[j];
		a[j] = temp;
	}

	public void printArray(int[] a) {
		for (int i = 0; i &lt; a.length; i++) {
			System.out.print(a[i] + "  ");
		}
		System.out.println();
	}
}

File: quick_sort.cpp
#include <stdio.h>
#include <conio.h>

int partition(int *a, int start, int end){
	int pivot=a[end];
	int pIndex=start;
	for (int i = start; i < end; ++i){
		if (a[i]<=pivot){
			int temp=a[i]; a[i]=a[pIndex]; a[pIndex]=temp;
			pIndex++;
		}
	}
	int temp = a[pIndex]; a[pIndex]=a[end]; a[end]=temp;;
	return pIndex;
}

void quicksort(int *a, int start, int end){
	if (start<end){
		int pIndex = partition(a, start, end);
		quicksort(a, start, pIndex-1);
		quicksort(a, pIndex+1, end);
	}
}

void show(int *a, int end){
	for(int i=0;i<end; i++)
    	printf("%d\t",a[i]);
}

int main(){
	int a[] = {2,1,3,8,5,7,6,4};
	int end = sizeof(a)/sizeof(a[0]);
	quicksort(a, 0, end-1);
    show(a,end);
    	
	getch();
	return 0;
}
